components/client/src/main/java/com/hotels/styx/client/healthcheck/UrlRequestHealthCheck.java

```diff
@@ -15,38 +15,37 @@
  */
 package com.hotels.styx.client.healthcheck;
 
-import com.codahale.metrics.Meter;
 import com.hotels.styx.api.HttpRequest;
-import com.hotels.styx.api.MetricRegistry;
 import com.hotels.styx.api.extension.Origin;
 import com.hotels.styx.client.HttpClient;
 import com.hotels.styx.common.SimpleCache;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
 
 import static com.hotels.styx.api.HttpHeaderNames.HOST;
 import static com.hotels.styx.api.HttpResponseStatus.OK;
 import static com.hotels.styx.client.healthcheck.OriginHealthCheckFunction.OriginState.HEALTHY;
 import static com.hotels.styx.client.healthcheck.OriginHealthCheckFunction.OriginState.UNHEALTHY;
-import static java.lang.String.format;
 
 /**
  * Health-check that works by making a request to a URL and ensuring that it gets an HTTP 200 OK code back.
  */
 public class UrlRequestHealthCheck implements OriginHealthCheckFunction {
-    private static final MeterFormat DEPRECATED_METER_FORMAT = new MeterFormat("origins.healthcheck.failure.%s");
-    private static final MeterFormat CORRECTED_METER_FORMAT = new MeterFormat("origins.%s.healthcheck.failure");
-
     private final String healthCheckUri;
-    private final SimpleCache<Origin, FailureMeter> meterCache;
+    private final SimpleCache<Origin, Counter> meterCache;
 
     /**
      * Construct an instance.
      *
      * @param healthCheckUri URI to make health-check requests to
-     * @param metricRegistry metric registry
+     * @param meterRegistry meter registry
      */
-    public UrlRequestHealthCheck(String healthCheckUri, MetricRegistry metricRegistry) {
+    public UrlRequestHealthCheck(String healthCheckUri, MeterRegistry meterRegistry) {
         this.healthCheckUri = uriWithInitialSlash(healthCheckUri);
-        this.meterCache = new SimpleCache<>(origin -> new FailureMeter(origin, metricRegistry));
+        this.meterCache = new SimpleCache<>(origin -> Counter.builder("origins.healthcheck.failures")
+        .tag("host", origin.host())
+        .tag("port", Integer.toString(origin.port()))
```

Comment:
Most other origin related metrics use the origin ID and application ID (both available on the origin) as tags, rather than host and port.